### PR TITLE
Support to expose operation metrics

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

We need operation metrics to know the operation details.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
cc @cfmcgrady @wForget @cxzl25

**Comment by turboFei** [comment]
cc @pan3793 have fixed the UT.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala [hunk 1]
```diff
@@ -53,6 +53,17 @@ abstract class KyuubiOperation(session: Session) extends AbstractOperation(sessi
 
   def remoteOpHandle(): TOperationHandle = _remoteOpHandle
 
+  @volatile protected var _fetchLogCount = 0L
+  @volatile protected var _fetchResultsCount = 0L
+
+  protected[kyuubi] def increaseFetchResultsCount(count: Int, fetchLog: Boolean): Unit = {
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala:59]: prefer to break it up, instead of distinguishing it by a flag

**Review by pan3793 — APPROVED** [review]
The feature makes sense to me

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala (hunk 1 above)
Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala]: done

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**